src/full/Agda/TypeChecking/CheckInternal.hs-boot

```diff
@@ -1,17 +1,45 @@
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE CPP            #-}
+
 module Agda.TypeChecking.CheckInternal where
 
+import qualified Data.Kind as Hs
+
 import Agda.Syntax.Internal
-import Agda.TypeChecking.Monad.Base (TCM)
+import Agda.TypeChecking.Monad
+import Agda.TypeChecking.Monad.Builtin (HasBuiltins)
+import Agda.TypeChecking.Monad.Statistics (MonadStatistics)
+import Agda.TypeChecking.Warnings
+import Agda.Utils.Except ( MonadError )
+
+type MonadCheckInternal m =
+  ( MonadReduce m
+  , MonadAddContext m
+  , MonadConstraint m
+  , MonadMetaSolver m
+  , MonadError TCErr m
+  , MonadWarning m
+  , MonadDebug m
+  , MonadStatistics m
+  , MonadFresh ProblemId m
+  , MonadFresh Int m
+  , HasBuiltins m
+  , HasConstInfo m
+  , HasOptions m
+#if MIN_VERSION_base(4,11,0)
+  , Fail.MonadFail m
+#endif
```

Comment:
The other checks around `MonadFail` were for `__GLASGOW_HASKELL__ >= 800` (whose first version of `base` is `base-4.9.0`), but this is against `base-4.11.0`. Are you sure this isn't supposed to be `__GLASGOW_HASKELL__ >= 800`, and in which case the condition can be removed now that 7.* support has been dropped?

(I'm just Cmd-F'ing for `#if` directives)